denops/gin/action/fixup.ts

```diff
@@ -67,3 +74,28 @@ async function doFixupInteractive(
     .dispatch("gin", "command", "", ["commit", `--fixup=${kind}:${commit}`])
     .catch((e) => console.error(`failed to execute git commit: ${e}`));
 }
+
+async function doFixupInstant(
+  denops: Denops,
+  bufnr: number,
+  range: Range,
+  gatherCandidates: GatherCandidates<Candidate>,
+): Promise<void> {
+  const xs = await gatherCandidates(denops, bufnr, range);
+  const commit = xs.map((v) => v.commit).join("\n");
+  await denops.dispatch("gin", "command", "", [
+    "commit",
+    `--fixup=${commit}`,
+  ]).then(() =>
```

Comment:
:question:

Why do you use `then` here?

Reply:
Because rebase should happen iff `commit --fixup` was successful.
For example, `commit --fixup` fails if there are no staged changes.

However, I noticed that this implementation continues to rebase anyway without changes.

I'm fine to split them into two awaits.